cmd/policy.go

```diff
@@ -70,42 +89,61 @@ func (sys *PolicySys) IsAllowed(args policy.Args) bool {
 	return args.IsOwner
 }
 
-// Init - initializes policy system from policy.json of all buckets.
-func (sys *PolicySys) Init(objAPI ObjectLayer) error {
-	if objAPI == nil {
-		return errInvalidArgument
-	}
-
+// Refresh PolicySys.
+func (sys *PolicySys) refresh(objAPI ObjectLayer) {
 	buckets, err := objAPI.ListBuckets(context.Background())
 	if err != nil {
-		return err
+		return
 	}
-
+	sys.RemoveDeletedBuckets(buckets)
 	for _, bucket := range buckets {
 		config, err := GetPolicyConfig(objAPI, bucket.Name)
 		if err != nil {
-			if _, ok := err.(BucketPolicyNotFound); !ok {
-				return err
+			if _, ok := err.(ObjectNotFound); ok {
```

Comment:
What is the difference between `ObjectNotFound` and `BucketPolicyNotFound`?

Reply:
`BucketPolicyNotFound` is the right error, fixed it